main.nf

```diff
@@ -873,9 +904,12 @@ process megahit {
 
     script:
     def input = params.single_end ? "-r \"${reads}\"" :  "-1 \"${reads[0]}\" -2 \"${reads[1]}\""
-    """
-    megahit -t "${task.cpus}" $input -o MEGAHIT --out-prefix "${name}"
-    """
+    if ( !params.megahit_fix_cpu_1 || task.cpus == 1 )
```

Comment:
I think this is not doing what it is supposed to do. I'd think what you want is
```
if ( params.megahit_fix_cpu_1 && task.cpus > 1 )
  error [...]
else
  """megahit [...]
```

Reply:
isn't that the same?

Reply:
Probably I misunderstood your aim here, but I thought you want to run megahit only when `params.megahit_fix_cpu_1=true` and `task.cpus=1` or when `params.megahit_fix_cpu_1=false` and `task.cpus>1`. In this case it does make a difference.


I tested with:
```
println "with &&:"
if ( megahit_fix_cpu_1 && cpus > 1 )
  println "error"
else
  println "megahit $cpus"
  
println "with ||:"
if ( !params.megahit_fix_cpu_1 || task.cpus == 1 )
  println "megahit $cpus"
else
  println "error"
```
with parameters:
```
megahit_fix_cpu_1=false
cpus=5
```
result:
```
with &&:
megahit 5
with ||:
megahit 5
```
with parameters:
```
megahit_fix_cpu_1=true
cpus=5
```
result:
```
with &&:
error
with ||:
megahit 5
```
--> here was the difference. And I thought that you aim for "error" here instead of running megahit.

with parameters:
```
megahit_fix_cpu_1=true
cpus=1
```
result:
```
with &&:
megahit 1
with ||:
megahit 1
```

Reply:
Yes, indeed I aim for an error in this case. In your test with "||" you wrote `task.cpus` instead of `cpus`. Could you try changing that?

Reply:
Alright, sorry, my mistake!